services/provider/server/server.go

```diff
@@ -969,6 +986,223 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 
 }
 
+// SHIM for GetStorageClaimConfig to return response to 4.18 clients
+func (s *OCSProviderServer) proxyGetStorageClaimConfigBlock(
+	req *pb.StorageClaimConfigRequest,
+	storageCluster *ocsv1.StorageCluster,
+	consumerConfig util.StorageConsumerResources,
+	kubeResources []client.Object,
+) ([]*pb.ExternalResource, error) {
+
+	var extR []*pb.ExternalResource
+
+	for _, resource := range kubeResources {
+		gvk, err := apiutil.GVKForObject(resource, s.scheme)
+		if err != nil {
+			return nil, err
+		}
+
+		switch gvk {
+		case v1.SchemeGroupVersion.WithKind("Secret"):
+			secret := resource.(*corev1.Secret)
+			provSecName := consumerConfig.GetCsiRbdProvisionerSecretName()
+			nodeSecName := consumerConfig.GetCsiRbdNodeSecretName()
```

Comment:
We already sent the entire kube resource list to this function, why do we need to look at the consume config again, or even send it to this function in the first place?


This comment is relevant for all of the other cases in this switch as well